File: app/src/main/java/jp/sourceforge/gokigen/memoma/dialogs/CreditDialog.java
package jp.sourceforge.gokigen.memoma.dialogs;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.Dialog;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;

import jp.sourceforge.gokigen.memoma.R;

/**
 *  クレジットを表示する
 * 
 * @author MRSa
 *
 */
public class CreditDialog
{
	private Activity context;

	/**
	 *   コンストラクタ
	 *
	 */
	public CreditDialog(Activity arg)
	{
		context = arg;
	}

    /**
     *   ダイアログを応答する
     *
     */
    public Dialog getDialog()
    {
    	LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    	if (inflater == null)
		{
			return (null);
		}
    	View layout = inflater.inflate(R.layout.creditdialog, null);  //  ?? http://www.mail-archive.com/android-developers@googlegroups.com/msg162003.html より
    	//View layout = inflater.inflate(R.layout.creditdialog,  context.findViewById(R.id.layout_root));

    	TextView text = layout.findViewById(R.id.creditmessage);
    	text.setText(context.getString(R.string.app_credit));
 //   	ImageView image = (ImageView) layout.findViewById(R.id.crediticon);
 //   	image.setImageResource(R.drawable.icon);

        AlertDialog.Builder builder = new AlertDialog.Builder(context);
        builder.setTitle(context.getString(R.string.app_name));
        builder.setIcon(R.drawable.icon);
        builder.setView(layout);
        builder.setCancelable(true);
        return (builder.create());
    }
}

File: app/src/main/java/jp/sourceforge/gokigen/memoma/dialogs/CreditDialog.kt
package jp.sourceforge.gokigen.memoma.dialogs

import android.content.Context
import android.view.LayoutInflater
import android.widget.TextView
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import jp.sourceforge.gokigen.memoma.R

class CreditDialog(private val myContext: Context) : DialogFragment()
{
    fun show()
    {
        val alertDialog = AlertDialog.Builder(myContext)
        try
        {
            val inflater = myContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            val layout = inflater.inflate(R.layout.creditdialog, null)
            val messageArea = layout.findViewById<TextView>(R.id.creditmessage)
            messageArea.text = myContext.getString(R.string.app_credit)
            alertDialog.setView(layout)
            alertDialog.setIcon(R.drawable.icon)
            alertDialog.setTitle(myContext.getString(R.string.app_name))
            alertDialog.setCancelable(true)
        }
        catch (e: Exception)
        {
            e.printStackTrace()
        }
        val alert = alertDialog.create()
        alert.show()
    }

    companion object
    {
        fun newInstance(context: Context): CreditDialog
        {
            return (CreditDialog(context))
        }
    }
}
